internal/authentication/tokengetter.go

```diff
@@ -86,7 +95,8 @@ func (t *TokenGetter) getToken(ctx context.Context, key types.NamespacedName) (*
 			Spec: authenticationv1.TokenRequestSpec{ExpirationSeconds: ptr.To(int64(t.expirationDuration / time.Second))},
 		}, metav1.CreateOptions{})
 	if err != nil {
-		return nil, err
+		saErr := &ServiceAccountNotFoundError{key.Name}
+		return nil, saErr
```

Comment:
I think we need to:
- Wrap this in a NotFound error check
- If the returned error is not a NotFound error, return the original error

This makes it so we are only returning our custom ServiceAccountNotFoundError whenever we know it is a NotFound error returned from our interaction with the Kubernetes API server. Not doing this means we assume that an error in this case always means the ServiceAccount is not found, which I think is a bad assumption to be making and will result in obfuscating any other errors that may be returned during this interaction.

Reply:
I rebuild and it seems to be working as expected:
```
status:
  conditions:
  - lastTransitionTime: "2024-11-26T16:08:21Z"
    message: 'Unable to authenticate with Kubernetes cluster using ServiceAccount
      "nginx-install": ServiceAccount "nginx-install" not found.'
    observedGeneration: 1
    reason: Failed
    status: Unknown
    type: Installed
  - lastTransitionTime: "2024-11-26T16:08:21Z"
    message: retrying to get installed bundle
    observedGeneration: 1
    reason: Retrying
    status: "True"
    type: Progressing